cmd/cluster-displayer/main.go

```diff
@@ -0,0 +1,199 @@
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %w", err)
+	}
+	if o.hiveKubeconfigPath == "" {
+		return fmt.Errorf("--hive-kubeconfig must be set")
+	}
+	return nil
+}
+
+type options struct {
+	logLevel           string
+	port               int
+	gracePeriod        time.Duration
+	hiveKubeconfigPath string
+}
+
+func addSchemes() error {
+	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
+		return fmt.Errorf("failed to add hivev1 to scheme: %w", err)
+	}
+	return nil
+}
+
+func getLabelsString(l map[string]string) string {
+	keys := make([]string, 0, len(l))
+	for k := range l {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var s []string
+	for _, k := range keys {
+		s = append(s, fmt.Sprintf("%s=%s", k, l[k]))
+	}
+	strings.Join(s, ",")
+	return strings.Join(s, ",")
+}
+
+func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client) (*Page, error) {
+	clusterPools := &hivev1.ClusterPoolList{}
+	if err := hiveClient.List(ctx, clusterPools); err != nil {
+		return nil, fmt.Errorf("failed to list cluster pools: %w", err)
+	}
+	var page Page
+	page.Data = []map[string]string{}
+	for _, p := range clusterPools.Items {
+		maxSize := "nil"
+		if p.Spec.MaxSize != nil {
+			maxSize = strconv.FormatInt(int64(*p.Spec.MaxSize), 10)
+		}
+		page.Data = append(
+			page.Data, map[string]string{
+				"namespace": p.Namespace,
+				"name":      p.Name,
+				"ready":     strconv.FormatInt(int64(p.Status.Ready), 10),
+				"size":      strconv.FormatInt(int64(p.Spec.Size), 10),
+				"maxSize":   maxSize,
+				"imageSet":  p.Spec.ImageSetRef.Name,
+				"labels":    getLabelsString(p.Labels),
+			},
+		)
+	}
+	return &page, nil
+}
+
+func getResource(ctx context.Context, hiveClient ctrlruntimeclient.Client, resource string) (*Page, error) {
+	switch strings.ToLower(resource) {
+	case "clusterpools":
+		return getClusterPoolPage(ctx, hiveClient)
+	default:
+		return nil, fmt.Errorf("unknown resource: %s", resource)
+	}
+}
+
+func getRouter(ctx context.Context, hiveClient ctrlruntimeclient.Client) *mux.Router {
+	handler := mux.NewRouter()
+
+	handler.HandleFunc("/api/health", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(map[string]bool{"ok": true})
+	}).Methods(http.MethodGet)
+
+	handler.HandleFunc("/api/v1/{resource}", func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		resource := vars["resource"]
+		logrus.WithField("resource", resource).Info("serving resource")
+		page, err := getResource(ctx, hiveClient, resource)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		callbackName := r.URL.Query().Get("callback")
+		bytes, err := json.Marshal(page)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		w.Header().Set("Content-Type", "application/javascript")
+		fmt.Fprintf(w, "%s(%s);", callbackName, string(bytes))
```

Comment:
`jsonp` support for ajax cross domain call.
We could use https://github.com/gin-gonic/gin#jsonp but this shows what is under the hood.

Reply:
Why do we need AJAX cross-domain calls?

Reply:
https://github.com/openshift/ci-docs/pull/154/files#diff-96dd75a968976edd5e03170268ed9085733f75c3fb24f992ae613c89e6de42dcR49

My chrome blocks it otherwise (if i use the normal json call) in the test.

Reply:
Forgot the main thing.
I want to embed the display on our doc site.

![Screen Shot 2021-06-16 at 8 42 35 AM](https://user-images.githubusercontent.com/4013349/122316940-a886ae80-ceea-11eb-99c8-77f4c2efd71e.png)

Reply:
![Screen Shot 2021-06-16 at 10 05 47 PM](https://user-images.githubusercontent.com/4013349/122319960-6a3fbe00-ceef-11eb-9103-3d7b3f1217ff.png)
It looks like this now.